cg/store/filters/status_case_filters.py

```diff
@@ -126,9 +126,37 @@ def filter_cases_by_customer_entry_id(cases: Query, customer_entry_id: int, **kw
     return cases.filter(Customer.id == customer_entry_id)
 
 
+def filter_cases_by_customer_entry_ids(
+    cases: Query, customer_entry_ids: List[int], **kwargs
+) -> Query:
+    """Return cases with matching customer ids."""
+    return cases.filter(Customer.id.in_(customer_entry_ids)) if customer_entry_ids else cases
+
+
+def filter_cases_by_action(cases: Query, action: str, **kwargs) -> Query:
+    """Return cases with matching action."""
+    return cases.filter(Family.action == action) if action else cases
+
+
 def filter_cases_by_name(cases: Query, name: str, **kwargs) -> Query:
     """Return cases with matching name."""
-    return cases.filter(Family.name == name)
+    return cases.filter(Family.name == name) if name else cases
+
+
+def filter_cases_by_matching_internal_id_or_name(
+    cases: Query, internal_id_search_pattern, name_search_pattern, **kwargs
+) -> Query:
+    """Return cases with matching internal id or name."""
+    return (
+        cases.filter(
+            or_(
+                Family.internal_id.like(f"%{internal_id_search_pattern}%"),
+                Family.name.like(f"%{name_search_pattern}%"),
+            )
+        )
+        if internal_id_search_pattern or name_search_pattern
+        else cases
+    )
```

Comment:
Can you split these in two? I think the way I did it in sample gets the same behaviour. without using or_ etc.
We should decide on one structure and keep it the same. This would also ensure that the filter function filters on only one thing instead of potentially 2.

Reply:
```Python
    def get_samples_by_customer_id_and_pattern(
        self, *, customers: Optional[List[Customer]] = None, pattern: str = None
    ) -> List[Sample]:
        """Get samples by customer and sample internal id  or sample name pattern."""
        samples: Query = self._get_query(table=Sample)
        customer_entry_ids = None
        filter_functions: List[SampleFilter] = []
        if customers:
            customer_entry_ids: List[int] = [customer.id for customer in customers]
            filter_functions.append(SampleFilter.FILTER_BY_CUSTOMER_ENTRY_IDS)
        if pattern:
            filter_functions.extend(
                [SampleFilter.FILTER_BY_INTERNAL_ID_PATTERN, SampleFilter.FILTER_BY_NAME_PATTERN]
            )
        filter_functions.append(SampleFilter.ORDER_BY_CREATED_AT_DESC)
        return apply_sample_filter(
            samples=samples,
            customer_entry_ids=customer_entry_ids,
            name_pattern=pattern,
            internal_id_pattern=pattern,
            filter_functions=filter_functions,
        ).all()
```

Reply:
see the if pattern:

Reply:
Hm, but this will not behave in the same way. It will be AND instead of OR.